operator-common/src/main/java/io/strimzi/operator/common/model/PemAuthIdentity.java

```diff
@@ -0,0 +1,113 @@
+                .ifPresent(objectMeta -> {
+                    secretName = objectMeta.getName();
+                    secretNamespace = objectMeta.getNamespace();
+                });
+        pemPrivateKey = Util.decodeFromSecret(secret, String.format("%s.key", secretKey));
+        pemCertificateChainBytes = Util.decodeFromSecret(secret, String.format("%s.crt", secretKey));
+        pemCertificateChain = validateCertificateChain(secretKey);
+    }
+
+    /**
+     * Create a new instance of PemAuthIdentity that represents the identity of the
+     * cluster operator during TLS client authentication. This also validates the provided
+     * Secret contains a valid certificate chain.
+     *
+     * @param secret Kubernetes Secret containing the client authentication identity
+     *
+     * @return PemAuthIdentity to use as the client authentication identity during TLS authentication
+     */
+    public static PemAuthIdentity clusterOperator(Secret secret) {
+        return new PemAuthIdentity(secret, "cluster-operator");
+    }
+
+    /**
+     * Create a new instance of PemAuthIdentity that represents the identity of the
+     * entity (i.e. user or topic) operator during TLS client authentication. This also validates
+     * the provided Secret contains a valid certificate chain.
+     *
+     * @param secret Kubernetes Secret containing the client authentication identity
+     *
+     * @return PemAuthIdentity to use as the client authentication identity during TLS authentication
+     */
+    public static PemAuthIdentity entityOperator(Secret secret) {
+        return new PemAuthIdentity(secret, "entity-operator");
+    }
+
+    /**
+     * @return The certificate chain for this authentication identity as a X509Certificate
+     */
+    public X509Certificate pemCertificateChain() {
+        return pemCertificateChain;
+    }
+
+    /**
+     * @return The certificate chain for this authentication identity as a byte array
+     */
+    public byte[] pemCertificateChainBytes() {
+        return pemCertificateChainBytes;
+    }
+
+    /**
+     * @return The certificate chain for this authentication identity as a String
+     */
+    public String pemCertificateChainString() {
+        return Util.decodeToString(pemCertificateChainBytes);
+    }
+
+    /**
+     * @return The private key for this authentication identity as a byte array
+     */
+    public byte[] pemPrivateKey() {
+        return pemPrivateKey;
+    }
+
+    /**
+     * @return The private key for this authentication identity as a String
+     */
+    public String pemPrivateKeyString() {
+        return Util.decodeToString(pemPrivateKey);
+    }
+
+    private X509Certificate validateCertificateChain(String secretKey) {
+        try {
+            final CertificateFactory certificateFactory = CertificateFactory.getInstance("X.509");
+            return (X509Certificate) certificateFactory.generateCertificate(new ByteArrayInputStream(pemCertificateChainBytes));
```

Comment:
What about the case where the chain contains multiple intermediate CA certificates? Then the result cannot be a single X509 certificate. And in that case we should validate that it issuer of one certificate is the subject of the next.